servers/src/main/java/tachyon/master/file/FileSystemMaster.java

```diff
@@ -1476,6 +1520,183 @@ public void setState(long fileId, SetStateOptions options)
     }
   }
 
+  /**
+   * Schedules a file for async persistence.
+   *
+   * @param the id of the file for persistence
+   * @throws FileDoesNotExistException when the file does not exist
+   */
+  public synchronized void scheduleAsyncPersistence(long fileId) throws FileDoesNotExistException {
+    // find the worker
+    long workerId = getWorkerStoringFile(fileId);
+
+    // update the state
+    synchronized (mInodeTree) {
+      Inode inode = mInodeTree.getInodeById(fileId);
+      inode.setPersistenceState(FilePersistenceState.SCHEDULED);
+    }
+
+    if (!mWorkerToAsyncPersistFile.containsKey(workerId)) {
+      mWorkerToAsyncPersistFile.put(workerId, Sets.<Long>newHashSet());
+    }
+    mWorkerToAsyncPersistFile.get(workerId).add(fileId);
+  }
+
+  /**
+   * Gets the worker where the given file is stored.
+   *
+   * @param fileId the file id
+   * @return the storing worker
+   * @throws FileDoesNotExistException when the file does not exist on any worker
+   */
+  private long getWorkerStoringFile(long fileId) throws FileDoesNotExistException {
+    List<Long> workers = Lists.newArrayList();
+    try {
+      for (FileBlockInfo fileBlockInfo : getFileBlockInfoList(fileId)) {
+        for (BlockLocation blockLocation : fileBlockInfo.blockInfo.locations) {
```

Comment:
There is no verification that all the blocks are in Tachyon storage. Is this correct?